be/src/storage/lake/persistent_index_memtable.cpp

```diff
@@ -89,36 +91,65 @@ Status PersistentIndexMemtable::replace(const Slice* keys, const IndexValue* val
     for (unsigned long idx : replace_idxes) {
         auto key = keys[idx].to_string();
         const auto value = values[idx];
-        std::list<IndexValueInfo> index_value_infos;
-        index_value_infos.emplace_front(version, value);
-        if (auto [it, inserted] = _map.emplace(key, index_value_infos); !inserted) {
+        std::list<IndexValueWithVer> index_value_vers;
+        index_value_vers.emplace_front(version, value);
+        if (auto [it, inserted] = _map.emplace(key, index_value_vers); !inserted) {
             update_index_value(&it->second, version, value);
         }
     }
     return Status::OK();
 }
 
-Status PersistentIndexMemtable::get(size_t n, const Slice* keys, IndexValue* values, KeyIndexesInfo* not_found,
+Status PersistentIndexMemtable::get(size_t n, const Slice* keys, IndexValue* values, std::set<KeyIndex>* not_founds,
                                     size_t* num_found, int64_t version) {
     size_t nfound = 0;
     for (size_t i = 0; i < n; ++i) {
         auto key = std::string_view(keys[i]);
         auto it = _map.find(key);
         if (it == _map.end()) {
             values[i] = NullIndexValue;
-            not_found->key_index_infos.emplace_back(i);
+            not_founds->insert(i);
         } else {
             // Assuming we want the latest (first) value due to emplace_front in updates/inserts
-            auto& index_value_infos = it->second;
-            auto old_index_value = index_value_infos.front().second;
-            values[i] = old_index_value;
-            nfound += old_index_value.get_value() != NullIndexValue;
+            auto& index_value_vers = it->second;
+            auto index_value = index_value_vers.front().second;
+            values[i] = index_value;
+            nfound += index_value.get_value() != NullIndexValue;
         }
     }
     *num_found = nfound;
     return Status::OK();
 }
 
+Status PersistentIndexMemtable::get(const Slice* keys, IndexValue* values, const std::set<KeyIndex>& key_indexes,
+                                    std::set<KeyIndex>* found_key_indexes, int64_t version) {
+    for (auto& key_index : key_indexes) {
+        auto key = std::string_view(keys[key_index]);
+        auto it = _map.find(key);
+        if (it != _map.end()) {
+            // Assuming we want the latest (first) value due to emplace_front in updates/inserts
+            auto& index_value_vers = it->second;
+            auto& index_value = index_value_vers.front().second;
+            values[key_index] = index_value.get_value();
+            found_key_indexes->insert(key_index);
+        }
+    }
+    return Status::OK();
+}
+
+size_t PersistentIndexMemtable::memory_usage() {
+    size_t mem_usage = 0;
+    for (auto const& it : _map) {
+        mem_usage += it.first.size();
```

Comment:
may worth considering the overhead of std::string object itself.

```suggestion
        mem_usage += it.first.size() + sizeof(std::string);
```